weasyl/character.py

```diff
@@ -234,15 +234,63 @@ def select_view(userid, charid, rating, ignore=True, anyway=None):
         "page_views": query[12],
         "friends_only": "f" in query[11],
         "hidden_submission": "h" in query[11],
-        # todo
-        "fave_count": define.execute("SELECT COUNT(*) FROM favorite WHERE (targetid, type) = (%i, 'f')",
-                                     [charid], ["element"]),
+        "fave_count": favorite.count(charid=charid),
         "comments": comment.select(userid, charid=charid),
         "sub_media": fake_media_items(charid, query[0], login, query[11]),
         "tags": searchtag.select(charid=charid),
     }
 
 
+def select_view_api(userid, charid, anyway=False, increment_views=False):
+    query = define.execute("""
```

Comment:
Could you use `define.engine.execute` instead, please? It would look something like this:

```python
query = define.engine.execute("""
    SELECT
        ch.userid, pr.username, ch.unixtime, ch.char_name, ch.age, ch.gender, ch.height, ch.weight, ch.species,
        ch.content, ch.rating, ch.settings, ch.page_views, pr.config
    FROM character ch
        INNER JOIN profile pr USING (userid)
    WHERE ch.charid = %(charid)s
""", charid=charid).fetchone()
```

Then you’ll be able to use `query.settings` instead of `query[11]` and so on below. (We’re trying to get rid of `define.execute`.)